source_me.sh

```diff
@@ -5,5 +5,12 @@ export C_INCLUDE_PATH=`pwd`/include:$C_INCLUDE_PATH
 export CPLUS_INCLUDE_PATH=`pwd`/include:$CPLUS_INCLUDE_PATH
 export INCLUDE_PATH=`pwd`/include:$INCLUDE_PATH
 export PATH=`pwd`/bin:$PATH
-export CC=$(which gcc)
-export CXX=$(which g++)
+if [[ "x$CC" == "x" ]]
```

Comment:
I guess there's the issue of whether the shell the user happens to be using
supports that test syntax, but we probably don't care about weird shells.

On Fri, Feb 26, 2016 at 10:47 AM, Adam Novak <anovak@soe.ucsc.edu> wrote:

> Because I don't know bash is why. That's probably a better way to do it.
>
> On Fri, Feb 26, 2016 at 1:19 AM, Maciek Smuga-Otto <
> notifications@github.com> wrote:
>
>> In source_me.sh
>> <https://github.com/vgteam/xg/pull/14#discussion_r54219978>:
>>
>> > @@ -5,5 +5,12 @@ export C_INCLUDE_PATH=`pwd`/include:$C_INCLUDE_PATH
>> >  export CPLUS_INCLUDE_PATH=`pwd`/include:$CPLUS_INCLUDE_PATH
>> >  export INCLUDE_PATH=`pwd`/include:$INCLUDE_PATH
>> >  export PATH=`pwd`/bin:$PATH
>> > -export CC=$(which gcc)
>> > -export CXX=$(which g++)
>> > +if [[ "x$CC" == "x" ]]
>>
>> Stupid question, but why not just [ -z "$CC" ] (assuming you're checking
>> if it's unset *or* empty) or, if you *really* don't want to clobber it
>> even if it's set to the empty string (although I don't think that's needed
>> here), [ -z ${CC+x} ]?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/vgteam/xg/pull/14/files#r54219978>.
>>
>
>